### PR TITLE
ht1 Diana

### PR DESCRIPTION
**Pull request by dianasinenchenko**

### CODE REVIEW

File: src/main/java/school/lemon/changerequest/java/introduction/hw1/Task10.java [hunk 1]
```diff
@@ -4,6 +4,20 @@
...
+                if (i==2)  return true;
+                n=n/10;
+
+            }
+            while (n < 1);
+
+
+        }
+
+       return false;
```
Comment (xSAVIKx) [inline src/main/java/school/lemon/changerequest/java/introduction/hw1/Task10.java:19]: All you need to solve this task is to think about :
1. how can you get rightmost digit in number ?
2. how can you remove rightmost digit from number ?

For example take number n=321:
1. get rightmost number of n -> 1
2. is 1 == 2 ? -> no
3. remove rightmost digit from n -> n=32
4. go to step 1.

And you should do this while n is greater than 0.

File: src/main/java/school/lemon/changerequest/java/introduction/hw1/Task3.java [hunk 2]
```diff
@@ -2,6 +2,9 @@
 
 public class Task3 {
     public static boolean isBetween(int a, int b, int c) {
-        return false;
+
+       if ( (a >= b) && (b >= c ) || (a <= b) && (b <= c))  return true;
+
+        else return false;
```
Comment (xSAVIKx) [inline src/main/java/school/lemon/changerequest/java/introduction/hw1/Task3.java]: You don't need `if` statement here, cause any comparison in Java will return boolean value for you.
So here you can just return result of your comparisons.

File: src/main/java/school/lemon/changerequest/java/introduction/hw1/Task5.java [hunk 3]
```diff
@@ -2,11 +2,22 @@
...
     public static double calculateA(double x, double y, double z) {
-        return 0;
+
+        return (2*(Math.sin(x-(Math.PI) / 6))*calculateB(z)) / (0.5 + Math.pow(Math.sin(y), 2));
     }
 
     private static double calculateB(double z) {
-        return 0;
+
+        return (1 + Math.pow(z, 2) / (3 + (Math.pow(z, 2)) / 5));
```
Comment (xSAVIKx) [inline src/main/java/school/lemon/changerequest/java/introduction/hw1/Task5.java]: it's good practice to save results of some calculations, so that your program don't need to calculate same things many times.
In following case, you can save result of `Math.pow(z,2)` into some handy variable and reuse it wherever you need.

File: src/main/java/school/lemon/changerequest/java/introduction/hw1/Task6.java [hunk 4]
```diff
@@ -1,11 +1,25 @@
...
+        if(n==1) return 1;
+        result = fact(n-1) * n;
+        return result;
+    }
+
+
     public static double calculateS(double x) {
-        return 0d;
+
+        return  1 + x + Math.pow(x,2) / fact(2) + Math.pow(x,3) / fact(3) + Math.pow(x,4) / fact(4) ;
```
Comment (xSAVIKx) [inline src/main/java/school/lemon/changerequest/java/introduction/hw1/Task6.java]: It's OK, but what if you'll need to calculate 10 more terms ?
Try to solve this task using loop.

File: src/main/java/school/lemon/changerequest/java/introduction/hw1/Task7.java [hunk 5]
```diff
@@ -3,6 +3,28 @@
...
+            case 5 : day_num = "Friday";
+            break;
+            case 6 : day_num = "Saturday";
+            break;
+            case 7 : day_num = "Sunday";
+            break;
+            default: day_num = "error";
+            break;
+        }
+return day_num;
```
Comment (xSAVIKx) [inline src/main/java/school/lemon/changerequest/java/introduction/hw1/Task7.java]: First of all, variable names in java should follow Java Code Convention, and naming for variables use camelStyle.
Second point is that you don't need variable here at all.
You can just use `return` statement in each `case`.

File: src/main/java/school/lemon/changerequest/java/introduction/hw1/Task8.java [hunk 6]
```diff
@@ -2,6 +2,14 @@
 
 public class Task8 {
     public static int calculateSum(int N) {
-        return 0;
+        int Sum = 0;
```
Comment (xSAVIKx) [inline src/main/java/school/lemon/changerequest/java/introduction/hw1/Task8.java]: variables should start with lower case letters, so here it should be `sum`

File: src/main/java/school/lemon/changerequest/java/introduction/hw1/Task9.java [hunk 7]
```diff
@@ -3,6 +3,17 @@
...
+
+        int i;
+        {
+            do {
+                i = n / 3;
+
+            }
+            while (i % 3 == 1);
+
+            return true;
```
Comment (xSAVIKx) [inline src/main/java/school/lemon/changerequest/java/introduction/hw1/Task9.java]: Spend some more time on this task.
You are almost right with this solution.

**Review by xSAVIKx — CHANGES_REQUESTED** [review]
Please check comments below.

File: src/main/java/school/lemon/changerequest/java/introduction/hw1/Task9.java [hunk 8]
```diff
@@ -2,18 +2,18 @@
 
 public class Task9 {
 
-    public static boolean isPowerOfThree(int n) {
+    public static boolean isPowerOfThree(float n) {
```
Comment (xSAVIKx) [inline src/main/java/school/lemon/changerequest/java/introduction/hw1/Task9.java:5]: Why did you change method signature ?

File: src/main/java/school/lemon/changerequest/java/introduction/hw1/Task6.java [hunk 9]
```diff
@@ -2,24 +2,32 @@
...
     public static double calculateS(double x) {
+        double sum = 0;
+
+        for (int i = 1; i < 5; i++) {
+
+
+            sum += Math.pow(x, i) / fact(i);
+        }
+
+        return sum;
```
Comment (xSAVIKx) [inline src/main/java/school/lemon/changerequest/java/introduction/hw1/Task6.java:25]: Check your implementation one more time - you've got failed tests. You have missed some terms in expression.

**Review by xSAVIKx — CHANGES_REQUESTED** [review]
Please fix your implementation

File: src/main/java/school/lemon/changerequest/java/introduction/hw1/Task10.java (hunk 1 above)
Comment (dianasinenchenko) [reply src/main/java/school/lemon/changerequest/java/introduction/hw1/Task10.java:19]: like this? 
https://github.com/dianasinenchenko/introduction.hw1/blob/94b65cdb91774687814ba410a8cc054f656d2dc4/src/main/java/school/lemon/changerequest/java/introduction/hw1/Task10.java

or maybe smth wrong

**Review by xSAVIKx — APPROVED** [review]

**Comment by xSAVIKx** [comment]
Looks OK.
